### DISCUSSION TITLE
What's the benifits on latest 0.50.x? How the no-id feature should be used?

### DISCUSSION THREAD

**Discussion by breakersun** [rokath/trice#388] [General]
Hello, noticed there is a `no-id` feature branch under development, then the v0.50.x release came out.
So I was wondering how `no-id` feature works and how it gonna improve user's workflow?

Sorry if this is a naive question, a pointing out or a hint will be appreciated.

**Comment by rokath** [comment]
Probably you mean v0.56.x, right? The noid branch was merged into master before reaching the final aim. The idea is: Run `trice z` as post-compile or pre-commit step, so that all IDs are removed from the source code.  When running `trice u` as post-pull and pre-compile step, all IDs are restored just for the compiling. This will be the users choice. The advantage is, that the same code is usable in different projects having maybe different ID policies. For that it is important that the IDs are stay the same on each place between 2 compilations of the same project. For unique trice format strings this is no big deal. But the same format string could occur several times in one file and inbetween some edit of that file is possible. For a good result the li.json file is used too. To get the things well done, first the test environment needs improvement,  what was partially done in the noids branch. Also the movement to `trice`, `Trice` and `TRice` was an important step. `TRICE` should normally not be used anymore because it needs still `id`, `Id` or `ID` after `trice z` to keep the stamp size information.  Also `TRICE` includes the executed code instead of performing a function call, what is important for the code size if there are many trices.
